website_product_subscription/models/subscription_request.py

```diff
@@ -11,7 +11,9 @@ class SubscriptionRequest(models.Model):
     @api.multi
     def validate_request(self):
         res = super(SubscriptionRequest, self).validate_request()
-        self.create_web_access()
+        for request in self:
+            if not request.subscriber._has_web_access():
+                request.create_web_access()
```

Comment:
I think you should remove this test: if the user already has a web access (from current subscription) and is gifted a new subscription, he will not receive the _existing user gift email_.